### ISSUE TITLE
consider/discuss plan of showing events either starting today or from 1 week ago + show next 3 weeks?

### ISSUE THREAD

**Issue by gregsadetsky** [gregsadetsky/nycnoise#218]
for @nycnoise 's consideration:

pros:
- less scrolling to get to current events - current info would be closer to the top of the page
- makes it much easier near the end of the month to see events for the next days - obviously, worst cases are always the last days of the current month where you absolutely have to go to the next month's page (i.e. scroll completely up to the calendar and click a date in the next month)

cons:
- changes site / what people are used to seeing
- a tad more complexity to handle in the top calendar - some dates (i.e. tomorrow) would link to the same page, while others (8 days ago in the same month) would have to link to the full month page

**Comment by nycnoise** [comment]
i am officially VERY INTO this idea of last wks + next 3 wks, especially after having the "future" setup in event adding for a bit...

**Comment by gregsadetsky** [comment]
some other rando thoughts...

- start listings at current day i.e. like https://www.mucnoise.com/ or https://bostonshows.org/ etc. ? pros: more immediate, no calendar clicking
- remove calendar completely - big screen saving especially on mobile
- add links (instead of cal) to previous / current / next month - i.e. equivalent to /2024-02/ - pros: these links take a lot less space.
- had another crazy idea: only load today + 7 days into the future (site will load extremely quickly), and add "infinite scroll" i.e. as you scroll, it just keeps adding more days...! with everything above i.e. no cal, and links to other months.

pros/cons for all of these approaches, and obviously a departure from what you have today. but the big bonus of having a database is that we can experiment with these pretty easily. and see which we like / you feel is good, etc.!

oh and also, I do think that moving content up would be good, again especially on mobile

compare the current mobile homepage and what it could be below:

<img width="412" alt="image" src="https://github.com/gregsadetsky/nycnoise/assets/1017304/53789653-0b88-4c62-932e-d918978421f0">

<img width="408" alt="image" src="https://github.com/gregsadetsky/nycnoise/assets/1017304/64fead14-9262-4eb9-b900-0315855dd8ba">

**Comment by nycnoise** [comment]
i think i actually need to poll people & find out how often people are checking for days *other than* day-of...

**Comment by gregsadetsky** [comment]
Yeah 100%. Default to today-at-the-top, and month names link to full calendars e.g. what we already show on /2024-03

Then we just decide whether we show 1-2 weeks (site will load in 50ms...) and then just have links to previous / current / next months in the footer and that's it.....???

Super easy lift

**Comment by gregsadetsky** [comment]
we're live with an experiment: https://nyc-noise.com/no_cal !

feel free to send people there/check it out and we can iterate from this?

small notes:
- I temporarily removed the mailing list subscription form, I suggest moving it under the burger menu (i.e. same place where the search is)
- you'll probably notice but the last/curr/next month links appears both at the top and bottom of the page
- just a reminder for myself - a little bit less margin-bottom on the bottom month links div
- I'd also suggest maybe removing the "pwa" i.e. "install nyc-noise" prompt since it takes so much space? and revisit it later?

**Comment by Saltzshaker** [comment]
did a newer exploration that keeps the beloved calendar with the following changes:
* add option to click and expand "past events" (for month?) but this is closed by default, starting at the current day
* add back/forward month navigation for calendar - especially helpful to navigate future month events!
* gray out past dates so more visually clear what is in current / future

![Image](https://github.com/user-attachments/assets/4ba5288a-6913-449b-a8b0-67578e7a9976)

**Comment by nycnoise** [comment]
omg just caught this & it is... possibly a solution that does not violate my Month Feelings??? is there a way to see what this would look like on a phone??? IS THIS ACTUALLY THE WAY FORWARD

minimum tho HELL YES to month w/ forward/back toggles, but also... this is maybe perfect??? i'm gonna sleep on my Month Feels & then will revisit but initial take is YAY!!!!

**Comment by gregsadetsky** [comment]
I think that we all agree that the "month bar" was overall a net positive and that we could add it to the current site without any real issue right? (keeping the cal and really everything else)

<img width="700" alt="Image" src="https://github.com/user-attachments/assets/33bd96db-6f81-4d2f-931d-aada7aafb0ff" />

**Comment by nycnoise** [comment]
yes!!! maybe with like, can the forward & backward months be abbreviated? just thinking in terms of space on phone...

**Comment by gregsadetsky** [comment]
we now seem pretty much all on the same page re:

- start index page listings at the "current" day (see below)
- have way of seeing past events of the current month (ie start on the 13th if it's the 13th, but have ability to see 1st to 12th)
- probably show today + 4 weeks into the future (otherwise, if we're showing "today until the end of the month", then on the ~30th of each month, the whole index page would be ~one day long... not great)
- keep the monthly calendar at the top - it's too cute/familiar.
- add month navigation (next/previous month)
- obviously keep all of the /yyyy-mm/ pages active/working and show the full entire month when going to those urls (really mainly the changes discussed here relate to the index page only)
- automatically open/expand/show the map for the current (i.e. first) day
- "current day" means whatever day it is but also a "day" goes until/starts at 5am i.e. if it's march 13th 3am, the march 12th events should still show on the calendar as the current/top day. once it's march 13th 5:30am, then start at that day ie march 13th should be the top/current day.

**Comment by gregsadetsky** [comment]
did almost everything listed here, except doing the "if it's 4am, 'today' is the previous day" bit which is super smart, but is wrecking havoc on my brain right now. we can get back to it though...! ! !

---

@all-contributors please add @skeryl for code
